libs/core.lua

```diff
@@ -505,57 +307,76 @@ local function makeCore(config)
-          elseif tag.type == "blob" then
-            importBlob(writer, "deps/" .. alias .. ".lua", tag.object)
-          end
+    local function importEntry(hash, path, mode)
+      local kind, value = assert(db.loadAny(hash))
+      mode = mode or modes[kind]
+      if mode == modes.tree then
+        if path then
+          writer:add(path .. "/", "")
+        end
+        for i = 1, #value do
+          local entry = value[i]
+          local newPath = path and path .. '/' .. entry.name or entry.name
+          importEntry(entry.hash, newPath, entry.mode)
+        end
+      elseif modes.isFile(mode) then
+        local base = path:match("^(.*)%.lua$")
+        if base then
+          -- TODO: add an option to disable this in case you want uncompiled lua files.
+          log("compiling", path)
+          local fn = assert(loadstring(value, "bundle:" .. path))
+          value = string.dump(fn)
         end
+        writer:add(path, value, 9)
+      else
+        error("Can't handle entry for " .. path)
       end
     end
 
+    importEntry(rootHash)
+
     assert(uv.fs_write(fd, writer:finalize(), binSize))
     uv.fs_close(fd)
     assert(uv.fs_rename(tempFile, target))
     log("done building", target)
   end
 
-
-  function core.make(path, target)
-    local fs, newPath = vfs(path)
-    local meta = pkg.query(fs, newPath)
-    if not meta then
-      error("Not a package at: " .. path)
+  local function defaultTarget(meta)
+    local target = meta.target or meta.name:match("[^/]+$")
+    if jit.os == "Windows" then
+      target = target .. ".exe"
     end
-    return realMake(fs, newPath, meta, target)
+    return target
   end
 
-  local aliases = {
-    "^github://([^/]+)/([^/@]+)/?@(.+)$", "https://github.com/%1/%2/archive/%3.zip",
-    "^github://([^/]+)/([^/]+)/?$", "https://github.com/%1/%2/archive/master.zip",
-    "^gist://([^/]+)/(.+)/?$", "https://gist.github.com/%1/%2.git",
-  }
-  core.urlAilases = aliases
+  function core.make(source, target)
+    local zfs
+    -- Use vfs so that source can be a zip file or a folder.
+    zfs, source = vfs(source)
+    local meta = queryFs(zfs, source)
+    target = target or defaultTarget(meta)
+    local kind, hash = assert(import(core.db, zfs, source, {"-" .. target}, true))
+    assert(kind == "tree", "Only tree packages are supported for now")
+    local deps = getInstalled(zfs, source)
+    calculateDeps(core.db, deps, meta.dependencies)
+    hash = installDeps(core.db, hash, deps)
+    return makeZip(hash, target)
+  end
+
+  local function makeGit(target, url)
+    local path = (url:match("([^/]+).git$") or target or "app") .. ".git-clone"
+    log("downloading repo", url)
+    local stdout, stderr, code, signal = exec("git", "clone", "--depth=1", "--recursive", url, path)
+    if code == 0 and signal == 0 then
+      core.make(path, target)
+    else
+      error("Problem cloning: " .. stdout .. stderr)
+    end
+    exec("rm", "-rf", path)
```

Comment:
Hmm. If I was reading this source code, I would take a long pause on this line. maybe we could write a `rm -rf` function instead of shelling out.